### PR TITLE
feat(queuehost): support for rabbitmq as queuehost

### PR DESCRIPTION
**Pull request by roleyfoley**
## Description
Initial support for the queue host component using AmazonMQ in aws.

## Motivation and Context
he queue host component is intended to support managed queue broker services which offer a platform to host queues as opposed to instances of queues like we have with Sqs already 

## How Has This Been Tested?
Tested locally

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions
- [x] None

## Checklist:
- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] None of the above.

### CODE REVIEW

File: aws/components/queuehost/state.ftl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                    "Type" : AWS_AMAZONMQ_BROKER_RESOURCE_TYPE,
+                    "Ports" : ports,
+                    "Monitored" : true
+                },
+                "sg" : {
+                    "Id" : securityGroupId,
+                    "Name" : core.FullName,
+                    "Type" : AWS_VPC_SECURITY_GROUP_RESOURCE_TYPE
+                },
+                "rootCredentials" : getComponentSecretResoures(
```
Comment (RossMurr4y) [inline aws/components/queuehost/state.ftl:30]: think this should be `getComponentSecretResources`

**Review by RossMurr4y — DISMISSED** [review]

**Comment by roleyfoley** [comment]
Now added support for saving a credentials embeded URL as an attribute. This allows for support for our existing components

File: aws/components/queuehost/setup.ftl [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                    linkTarget=linkTarget
+                    inboundPorts=[ port ]
+                /]
+            [/#if]
+
+        [/#if]
+    [/#list]
+
+    [#-- Secret Management --]
+    [#local secretStoreLink = getLinkTarget(occurrence, solution.RootCredentials.SecretStore) ]
```
Comment (ml019) [inline aws/components/queuehost/setup.ftl:74]: Need to check the result to ensure link resolved correctly

File: aws/components/queuehost/setup.ftl [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    [#local passwordSecretKey = "password" ]
+
+    [@setupComponentSecret
```
Comment (ml019) [inline aws/components/queuehost/setup.ftl:77]: needs to be within a subset?

File: aws/services/amazonmq/resource.ftl [hunk 4]
```diff
@@ -6,7 +6,17 @@
             "UseRef" : true
         },
         DNS_ATTRIBUTE_TYPE : {
-            "Attribute" : "AmqpEndpoints"
+            "Value" : {
+                "Fn::Join" : [
+                    ";",
+                    {
+                        "Fn::GetAtt" : [
+                            "mqBrokerXmgmtXrabbithutch",
```
Comment (ml019) [inline aws/services/amazonmq/resource.ftl:14]: No other reference to this in the PR. I was not expecting anything in the master data for this ?

**Review by ml019 — CHANGES_REQUESTED** [review]

File: aws/components/queuehost/setup.ftl (hunk 2 above)
Comment (roleyfoley) [reply aws/components/queuehost/setup.ftl:74]: Fixed

File: aws/components/queuehost/setup.ftl (hunk 3 above)
Comment (roleyfoley) [reply aws/components/queuehost/setup.ftl:77]: The setup macro handles subsets as the default requires both a prologue and template subset

**Comment by roleyfoley** [comment]
@ml019 Should be good to go now

File: aws/deploymentframeworks/cf/output.ftl [hunk 5]
```diff
@@ -201,21 +201,41 @@
...
-                ((value.UseRef)!false)?then(
-                    {
-                        "Ref" : id
-                    },
-                    value.Value?has_content?then(
-                        value.Value,
+
+            [#if value.Replace?has_content ]
+                [#local content = getJSON(value.Replace)]
+                [#list [ "_id_" ] as replaceSting ]
```
Comment (ml019) [inline aws/deploymentframeworks/cf/output.ftl:207]:
```suggestion
                [#list [ "_id_" ] as replaceString ]
```

File: aws/deploymentframeworks/cf/output.ftl [hunk 6]
```diff
@@ -201,21 +201,41 @@
...
+                    [#switch replaceSting ]
```
Comment (ml019) [inline aws/deploymentframeworks/cf/output.ftl:208]:
```suggestion
                    [#switch replaceString ]
```

File: aws/deploymentframeworks/cf/output.ftl [hunk 7]
```diff
@@ -201,21 +201,41 @@
...
+                        [#case "_id_" ]
+                            [#local content = content?replace(replaceSting, id )]
```
Comment (ml019) [inline aws/deploymentframeworks/cf/output.ftl:210]:
```suggestion
                            [#local content = content?replace(replaceString, id )]
```

**Review by ml019 — CHANGES_REQUESTED** [review]

**Review by ml019 — APPROVED** [review]

**Pull request merged**